src/UnoFlipModel.java

```diff
@@ -382,151 +242,269 @@ private boolean isWinner(Player player){
-                    currentTurn = (currentTurn - 1 + numPlayers) % numPlayers;
-                    nextPlayerIndex = (currentTurn - 1 + numPlayers) % numPlayers;
+
+            //if next player's turn is being skipped (ex. player 2 is being skipped:  0->1->3->0)
+            if (this.skipTurn){
+                //clockwise (ex. 0->1->2->3)
+                if (this.turnDirection) {
+                    this.currentTurn = (this.currentTurn + 1) % this.numPlayers;
+                    this. nextPlayerIndex = (this.currentTurn + 1) % this.numPlayers;
+
+                //counterclockwise (ex. 0->3->2->1)
+                } else {
+                    this.currentTurn = (this.currentTurn - 1 + this.numPlayers) % this.numPlayers;
+                    this.nextPlayerIndex = (this.currentTurn - 1 + this.numPlayers) % this.numPlayers;
                 }
-                skipTurn = false;
+                this.skipTurn = false;
             }
-            status = " ";
+            this.status = STATUS_STANDARD;
             notifyViews();
-            turnFinished = false; // reset for next player
-        } else { // if player tries to skip turn
-            status = "CANNOT SKIP A TURN, EITHER PLAY A CARD FROM THE HAND OR DRAW FROM THE DECK";
+            this.turnFinished = false; // reset for next player
+
+        // if player tries to skip turn
+        } else {
+            this.status = STATUS_PLAYER_SKIPPING_TURN;
             notifyViews();
         }
 
     }
 
+    /**
+     * Returns the score of the winner, score is generated based on the cards opponents are left holding
+     * @return The score of the winner
+     */
+    private int getWinnerScore(){
+        int winnerScore = 0;
+
+        //handling if the last card played is a draw card (ex. RED_DRAW_ONE or WILD_DRAW_2)
+        //cards must still be given to next players before winner's score is calculated
+        if(this.topCard.getRank().ordinal() == Card.RANK_DRAW_ONE){
+            drawNCards(1,this.nextPlayerIndex);
+        } else if(this.topCard.getRank().ordinal() == Card.RANK_WILD_DRAW_2){
+            drawNCards(2,this.nextPlayerIndex);
+        }
+
+        //accumulate points
+        for(Player p: this.players){
+            winnerScore += p.getHandScore();
+        }
+        return winnerScore;
+    }
+
+    /**
+     * Checks to see if there is a playable card in hand before allowing player to draw a card from the deck
+     * @return return true if valid to draw a card from deck, otherwise false.
+     */
+    private boolean validSelfDrawOne(){
+        for (int i = 0; i < this.getCurrentPlayer().getHandSize();i++){
+            if (this.getCurrentPlayer().getCard(i).getRank() == this.getCurrentRank() || this.getCurrentPlayer().getCard(i).getColour() == this.getCurrentColour()){
+                return false;
+            }
+        }
+        return true;
+    }
 
     /**
      * Skip the turn of the next player
      */
     public void skipTurn(){
```

Comment:
I think a better name is setSkipTurnFlag() because skipTurn() seems like it actually skips the turn (which it doesn't) and setSkipTurn() should allow parameter to be true or false, but for this method we only want to set it to true each time, so that leaves us with setSkipTurnFlag()